contracts/bonding/libraries/TranscoderPool.sol

```diff
@@ -0,0 +1,372 @@
+     * @param _worseTranscoder Address of a transcoder with less stake than the updated transcoder
+     * @param _betterTranscoder Address of a transcoder with more stake than the updated transcoder
+     */
+    function decreaseTranscoderStake(Data storage self, address _transcoder, uint256 _amount, address _worseTranscoder, address _betterTranscoder) public {
+        // List must contain the transcoder
+        require(contains(self, _transcoder));
+
+        // Compute new stake
+        uint256 newStake = self.nodes[_transcoder].stake.sub(_amount);
+        // Remove transcoder from the list
+        removeTranscoder(self, _transcoder);
+
+        if (newStake > 0) {
+            // Only add transcoder if its new stake is non-zero
+            // Insert transcoder into the list with its new stake
+            addTranscoder(self, _transcoder, newStake, _worseTranscoder, _betterTranscoder);
+        }
+    }
+
+    /*
+     * @dev Checks if the list contains a transcoder
+     * @param _transcoder Address of transcoder
+     */
+    function contains(Data storage self, address _transcoder) public view returns (bool) {
+        return self.head == _transcoder || self.tail == _transcoder || self.nodes[_transcoder].prevTranscoder != address(0) || self.nodes[_transcoder].nextTranscoder != address(0);
+    }
+
+    /*
+     * @dev Checks if the list is full
+     */
+    function isFull(Data storage self) public view returns (bool) {
+        return self.size == self.maxSize;
+    }
+
+    /*
+     * @dev Checks if the list is empty
+     */
+    function isEmpty(Data storage self) public view returns (bool) {
+        return self.size == 0;
+    }
+
+    /*
+     * @dev Returns the current size of the list
+     */
+    function getSize(Data storage self) public view returns (uint256) {
+        return self.size;
+    }
+
+    /*
+     * @dev Returns the maximum size of the list
+     */
+    function getMaxSize(Data storage self) public view returns (uint256) {
+        return self.maxSize;
+    }
+
+    /*
+     * @dev Returns the stake of a transcoder in the list
+     * @param _transcoder Address of transcoder
+     */
+    function getTranscoderStake(Data storage self, address _transcoder) public view returns (uint256) {
+        return self.nodes[_transcoder].stake;
+    }
+
+    /*
+     * @dev Returns the transcoder in the list with the least stake (the head)
+     */
+    function getWorstTranscoder(Data storage self) public view returns (address) {
+        return self.head;
+    }
+
+    /*
+     * @dev Returns the transcoder in the list with the most stake (the tail)
+     */
+    function getBestTranscoder(Data storage self) public view returns (address) {
+        return self.tail;
+    }
+
+    /*
+     * @dev Returns the previous transcoder in the list for a transcoder
+     * @param _transcoder Address of transcoder
+     */
+    function getNextTranscoder(Data storage self, address _transcoder) public view returns (address) {
```

Comment:
`getNext` and `getPrev` are very confusing, especially since the comments mix prev + next.